### PR TITLE
MNG-6605 Unable to suppress download messages in interactive mode

### PR DESCRIPTION
**Pull request by rotty3000**
A solution for quieting the `Download[ed|ing] from ...` messages.

This solution introduces the system property flag `-Dorg.apache.maven.cli.transfer.ConsoleMavenTransferListener.quiet` which can either be used on the cli or be set in the `MAVEN_OPTS` so that it affects system wide.

Signed-off-by: Raymond Auge <rotty3000@apache.org>

Following this checklist to help us incorporate your 
contribution quickly and easily:

 - [x] Make sure there is a [JIRA issue](https://issues.apache.org/jira/browse/MNG) filed 
       for the change (usually before you start working on it).  Trivial changes like typos do not 
       require a JIRA issue.  Your pull request should address just this issue, without 
       pulling in other changes.
 - [x] Each commit in the pull request should have a meaningful subject line and body.
 - [x] Format the pull request title like `[MNG-XXX] - Fixes bug in ApproximateQuantiles`,
       where you replace `MNG-XXX` with the appropriate JIRA issue. Best practice
       is to use the JIRA issue title in the pull request title and in the first line of the 
       commit message.
 - [x] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
 - [x] Run `mvn clean verify` to make sure basic checks pass. A more thorough check will 
       be performed on your pull request automatically.
 - [x] You have run the [Core IT][core-its] successfully.

If your pull request is about ~20 lines of code you don't need to sign an
[Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf) if you are unsure
please ask on the developers list.

To make clear that you license your contribution under 
the [Apache License Version 2.0, January 2004](http://www.apache.org/licenses/LICENSE-2.0)
you have to acknowledge this by using the following check-box.

 - [x] I hereby declare this contribution to be licenced under the [Apache License Version 2.0, January 2004](http://www.apache.org/licenses/LICENSE-2.0)

 - [x] In any other case, please file an [Apache Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

[core-its]: https://maven.apache.org/core-its/core-it-suite/

### CODE REVIEW

File: maven-embedder/src/main/java/org/apache/maven/cli/CLIManager.java [hunk 1]
```diff
@@ -101,6 +101,8 @@
 
     public static final String BUILDER = "b";
 
+    public static final String NO_TRANSFER_PROGRESS = "np";
```
Comment (michael-o) [inline maven-embedder/src/main/java/org/apache/maven/cli/CLIManager.java]: This one's to similar to npu, npr and friends. It should be ntp.

File: maven-embedder/src/main/java/org/apache/maven/cli/CLIManager.java [hunk 2]
```diff
@@ -140,6 +142,7 @@ public CLIManager()
         options.addOption( OptionBuilder.withLongOpt( "threads" ).hasArg().withDescription( "Thread count, for instance 2.0C where C is core multiplied" ).create( THREADS ) );
         options.addOption( OptionBuilder.withLongOpt( "legacy-local-repository" ).withDescription( "Use Maven 2 Legacy Local Repository behaviour, ie no use of _remote.repositories. Can also be activated by using -Dmaven.legacyLocalRepo=true" ).create( LEGACY_LOCAL_REPOSITORY ) );
         options.addOption( OptionBuilder.withLongOpt( "builder" ).hasArg().withDescription( "The id of the build strategy to use" ).create( BUILDER ) );
+        options.addOption( OptionBuilder.withLongOpt( "no-transfer-progress" ).withDescription( "Do not display transfer progress when downloading" ).create( NO_TRANSFER_PROGRESS ) );
```
Comment (michael-o) [inline maven-embedder/src/main/java/org/apache/maven/cli/CLIManager.java]: It is uploading too

**Comment by rotty3000** [comment]
Ah ok, I even had ntp but changed it because I thought it might imply
something about time but I'll change it back.

On Sun, Mar 31, 2019, 17:22 Michael Osipov, <notifications@github.com>
wrote:

> *@michael-o* requested changes on this pull request.
>
> Except for those nags, the rest looks fine to me.
> ------------------------------
>
> In maven-embedder/src/main/java/org/apache/maven/cli/CLIManager.java
> <https://github.com/apache/maven/pull/239#discussion_r270685287>:
>
> > @@ -101,6 +101,8 @@
>
>      public static final String BUILDER = "b";
>
> +    public static final String NO_TRANSFER_PROGRESS = "np";
>
> This one's to similar to npu, npr and friends. It should be ntp.
> ------------------------------
>
> In maven-embedder/src/main/java/org/apache/maven/cli/CLIManager.java
> <https://github.com/apache/maven/pull/239#discussion_r270685296>:
>
> > @@ -140,6 +142,7 @@ public CLIManager()
>          options.addOption( OptionBuilder.withLongOpt( "threads" ).hasArg().withDescription( "Thread count, for instance 2.0C where C is core multiplied" ).create( THREADS ) );
>          options.addOption( OptionBuilder.withLongOpt( "legacy-local-repository" ).withDescription( "Use Maven 2 Legacy Local Repository behaviour, ie no use of _remote.repositories. Can also be activated by using -Dmaven.legacyLocalRepo=true" ).create( LEGACY_LOCAL_REPOSITORY ) );
>          options.addOption( OptionBuilder.withLongOpt( "builder" ).hasArg().withDescription( "The id of the build strategy to use" ).create( BUILDER ) );
> +        options.addOption( OptionBuilder.withLongOpt( "no-transfer-progress" ).withDescription( "Do not display transfer progress when downloading" ).create( NO_TRANSFER_PROGRESS ) );
>
> It is uploading too
>
Ah right. Will update later today.

Thanks for the feedback.

**Comment by rotty3000** [comment]
Updated as requested!

**Comment by jira-importer** [comment]
Resolve #8057

**Pull request closed without merging**